### PR TITLE
Update Crux Dockerfiles to use pre-built solvers

### PR DESCRIPTION
**Pull request by atomb**
Also update freeze files to work with latest code and use prebuilt solvers in other workflows.

### CODE REVIEW

**Review by kquick — APPROVED** [review]
I have a couple of concerns noted above, but otherwise this looks good.

File: .github/Dockerfile-crux-llvm [hunk 1]
```diff
@@ -1,71 +1,51 @@
...
-ARG DIR=/crux-llvm
-COPY --from=solvers /solvers/rootfs /
-COPY --from=build /usr/local/bin/* /usr/local/bin/
+COPY --from=build /usr/local/bin/* /usr/local/bin
 
+ARG DIR=/crux-llvm
 WORKDIR ${DIR}
 ADD crux-llvm/c-src c-src
-RUN cp c-src/libcxx-7.1.0.bc c-src/libcxx-7.0.1.bc
+RUN cp c-src/libcxx-9.0.0.bc c-src/libcxx-11.0.1.bc
```
[the remark this answers is not in the document]
Comment (atomb) [reply .github/Dockerfile-crux-llvm:46]: We haven't created builds of the libcxx code for LLVM 11, so this is an admittedly awkward workaround. Once we've done the work to build libcxx for each LLVM version, we could use the LLVM 11 version directly.

File: .github/workflows/crucible-wasm-build.yml [hunk 2]
```diff
@@ -74,8 +74,8 @@ jobs:
       - shell: bash
         run: .github/ci.sh install_system_deps
         env:
-          Z3_VERSION: "4.8.8"
-          YICES_VERSION: "2.6.2"
+          SOLVER_PKG_VERSION: "snapshot-20210917"
+          BIN_ZIP_FILE: ${{ matrix.os }}-bin.zip
```
[the remark this answers is not in the document]
Comment (atomb) [reply .github/workflows/crucible-wasm-build.yml:78]: Yeah, that's a good point. I'll tweak it to do it that way.

File: .github/Dockerfile-crux-llvm (hunk 1 above)
[the remark this answers is not in the document]
Comment (kquick) [reply .github/Dockerfile-crux-llvm:46]: Understood.  It would probably be good to add a comment to explain this situation.

**Pull request merged**